components/raftstore/src/store/util.rs

```diff
@@ -286,6 +292,35 @@ pub fn compare_region_epoch(
     Ok(())
 }
 
+pub fn find_sibling_regions(
+    store_meta: &Arc<Mutex<StoreMeta>>,
+    cfg: &Config,
+    cop_cfg: &CopConfig,
+    region: &Region,
+) -> Vec<Region> {
+    let meta = store_meta.lock().unwrap();
+    if cfg.right_derive_when_split {
+        meta.region_ranges
+            .range((Unbounded::<Vec<u8>>, Excluded(enc_end_key(region))))
+            .rev()
+            .take(cop_cfg.batch_split_limit as usize)
+            .map(|(_, region_id)| meta.regions[region_id].to_owned())
+            .filter(|r| {
+                compare_region_epoch(r.get_region_epoch(), region, true, true, false).is_ok()
+            })
+            .collect()
+    } else {
+        meta.region_ranges
+            .range((Excluded(enc_end_key(region)), Unbounded::<Vec<u8>>))
```

Comment:
Should it be `enc_start_key`?

Reply:
The region itself should not be included here.